man pages: fix inconsistencies

edsantiago opened the pull request:
I wrote a script to cross-reference podman --help against
man pages. It found a bunch of inconsistencies fix them:

 * options missing from man pages
 * options misspelled or misformatted in man pages (usually
   misplaced asterisks or missing dashes, but see --dns-opt)
 * one spurious comma in the actual source file --help

This is a fix in which I iterate over 'podman CMD --help'
and check for presence in man pages. The other way around
(look for flags in man pages, check podman CMD --help)
is probably impossible: there are too many special cases

Signed-off-by: Ed Santiago <santiago@redhat.com>

edsantiago commented:
The rest of the output, which I'm probably going to give up on:
```
podman-help-man-xref: podman build: --cni-config-dir in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman build: --net in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman build: --runtime in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman commit: --iidfile in docs/source/markdown/podman-commit.1.md, but not --help
podman-help-man-xref: podman container commit: --iidfile in docs/source/markdown/podman-commit.1.md, but not --help
podman-help-man-xref: podman container create: --cpu-count in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --ip6 in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --link-local-ip in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --net in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --network-alias in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container inspect: --type in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman container inspect: -t in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman container list: --namespace in docs/source/markdown/podman-ps.1.md, but not --help
podman-help-man-xref: podman container run: --ip6 in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --link-local-ip in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --net in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --network-alias in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container runlabel: --rootfs in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container runlabel: --storage in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container runlabel: -n in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container commit: --iidfile in docs/source/markdown/podman-commit.1.md, but not --help
podman-help-man-xref: podman container create: --cpu-count in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --ip6 in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --link-local-ip in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --net in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --network-alias in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container inspect: --type in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman container inspect: -t in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman container list: --namespace in docs/source/markdown/podman-ps.1.md, but not --help
podman-help-man-xref: podman container run: --ip6 in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --link-local-ip in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --net in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --network-alias in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container runlabel: --rootfs in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container runlabel: --storage in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container runlabel: -n in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman create: --cpu-count in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman create: --ip6 in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman create: --link-local-ip in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman create: --net in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman create: --network-alias in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman image build: --cni-config-dir in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image build: --net in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image build: --runtime in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image inspect: --latest in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: --size in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: --type in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -l in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -s in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -t in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image trust show: --pubkeysfile in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --type in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -f in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -t in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --pubkeysfile in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --type in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -f in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -t in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image build: --cni-config-dir in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image build: --net in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image build: --runtime in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image inspect: --latest in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: --size in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: --type in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -l in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -s in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -t in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image trust show: --pubkeysfile in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --type in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -f in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -t in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --pubkeysfile in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --type in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -f in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -t in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman network create --help has --driver, but --driver not in docs/source/markdown/podman-network-create.1.md
podman-help-man-xref: podman network create --help has --driver, but --driver not in docs/source/markdown/podman-network-create.1.md
podman-help-man-xref: podman pod prune --help has -f, but -f not in docs/source/markdown/podman-pod-prune.1.md
podman-help-man-xref: podman pod ps --help has -l, but -l not in docs/source/markdown/podman-pod-ps.1.md
podman-help-man-xref: podman pod prune --help has -f, but -f not in docs/source/markdown/podman-pod-prune.1.md
podman-help-man-xref: podman pod ps --help has -l, but -l not in docs/source/markdown/podman-pod-ps.1.md
podman-help-man-xref: podman ps: --namespace in docs/source/markdown/podman-ps.1.md, but not --help
podman-help-man-xref: podman run: --ip6 in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman run: --link-local-ip in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman run: --net in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman run: --network-alias in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman build: --cni-config-dir in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman build: --net in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman build: --runtime in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman commit: --iidfile in docs/source/markdown/podman-commit.1.md, but not --help
podman-help-man-xref: podman container commit: --iidfile in docs/source/markdown/podman-commit.1.md, but not --help
podman-help-man-xref: podman container create: --cpu-count in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --ip6 in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --link-local-ip in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --net in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --network-alias in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container inspect: --type in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman container inspect: -t in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman container list: --namespace in docs/source/markdown/podman-ps.1.md, but not --help
podman-help-man-xref: podman container run: --ip6 in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --link-local-ip in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --net in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --network-alias in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container runlabel: --rootfs in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container runlabel: --storage in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container runlabel: -n in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container commit: --iidfile in docs/source/markdown/podman-commit.1.md, but not --help
podman-help-man-xref: podman container create: --cpu-count in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --ip6 in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --link-local-ip in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --net in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container create: --network-alias in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman container inspect: --type in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman container inspect: -t in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman container list: --namespace in docs/source/markdown/podman-ps.1.md, but not --help
podman-help-man-xref: podman container run: --ip6 in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --link-local-ip in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --net in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container run: --network-alias in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman container runlabel: --rootfs in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container runlabel: --storage in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman container runlabel: -n in docs/source/markdown/podman-container-runlabel.1.md, but not --help
podman-help-man-xref: podman create: --cpu-count in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman create: --ip6 in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman create: --link-local-ip in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman create: --net in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman create: --network-alias in docs/source/markdown/podman-create.1.md, but not --help
podman-help-man-xref: podman image build: --cni-config-dir in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image build: --net in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image build: --runtime in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image inspect: --latest in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: --size in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: --type in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -l in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -s in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -t in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image trust show: --pubkeysfile in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --type in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -f in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -t in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --pubkeysfile in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --type in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -f in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -t in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image build: --cni-config-dir in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image build: --net in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image build: --runtime in docs/source/markdown/podman-build.1.md, but not --help
podman-help-man-xref: podman image inspect: --latest in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: --size in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: --type in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -l in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -s in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image inspect: -t in docs/source/markdown/podman-inspect.1.md, but not --help
podman-help-man-xref: podman image trust show: --pubkeysfile in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --type in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -f in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -t in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --pubkeysfile in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: --type in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -f in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman image trust show: -t in docs/source/markdown/podman-image-trust.1.md, but not --help
podman-help-man-xref: podman network create --help has --driver, but --driver not in docs/source/markdown/podman-network-create.1.md
podman-help-man-xref: podman network create --help has --driver, but --driver not in docs/source/markdown/podman-network-create.1.md
podman-help-man-xref: podman pod prune --help has -f, but -f not in docs/source/markdown/podman-pod-prune.1.md
podman-help-man-xref: podman pod ps --help has -l, but -l not in docs/source/markdown/podman-pod-ps.1.md
podman-help-man-xref: podman pod prune --help has -f, but -f not in docs/source/markdown/podman-pod-prune.1.md
podman-help-man-xref: podman pod ps --help has -l, but -l not in docs/source/markdown/podman-pod-ps.1.md
podman-help-man-xref: podman ps: --namespace in docs/source/markdown/podman-ps.1.md, but not --help
podman-help-man-xref: podman run: --ip6 in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman run: --link-local-ip in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman run: --net in docs/source/markdown/podman-run.1.md, but not --help
podman-help-man-xref: podman run: --network-alias in docs/source/markdown/podman-run.1.md, but not --help
```
Most of those are bogus, e.g. man page entries that simply say "Not implemented" (--ip6), or the giant mess in `podman image trust` (nonstandard man page), or the `-l` and `-s` options in `podman inspect`. Some are complete stumpers (`podman commit --iidfile`, `podman create --cpu-count`  ???). Some I think are real errors: `podman runlabel -n`.

mheon commented:
/approve
LGTM

TomSweeneyRedHat commented:
LGTM, nice work @edsantiago !

rhatdan commented on docs/source/markdown/podman.1.md:
```diff
@@ -116,6 +116,10 @@ Path to the tmp directory, for libpod runtime content.
 
 NOTE --tmpdir is not used for the temporary storage of downloaded images.  Use the environment variable `TMPDIR` to change the temporary storage location of downloaded container images. Podman defaults to use `/var/tmp`.
 
+**--trace**
```
In another PR we were hiding --trace, so this should NOT be documented.

rhatdan replied:
https://github.com/containers/libpod/pull/5402

edsantiago commented:
New force-push. Removed `--trace` from `podman.1.md`. Fixed a few more issues detected by new version of script (podman network create, pod prune, pod ps)

rhatdan commented:
LGTM

rhatdan commented:
/lgtm
/hold

edsantiago commented:
/hold cancel
